src/ast.ls

```diff
@@ -837,6 +836,13 @@ class exports.Chain extends Node
                     i = 0
                 call <<< method: \.apply, args: [ctx or Literal \null; JS args]
 
+    expand-vivify: (o) !->
+        {tails} = this
+        i = 0
+        while i < tails.length when delete tails[i++].vivify
+            @head = Assign Chain(@head, tails.splice 0, i), that!, \= \||
+            i = 0
```

Comment:
`tails` can contain multiple things in a row that aren't vivifications. The `when` guard makes `i` keep incrementing until it finds a vivification, at which point the loop body `splice`s out everything in `tails` up to that vivification into the `Assign`. `i` needs to be reset back to `0` at that point because `i` elements have been removed from `tails`.